src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -48,85 +48,147 @@ module Config : sig
   val path : t -> string
   val of_string : string -> t
   val of_flags : string list -> t
-  val to_flags : t -> string list
+
+  (** [recent] should be true if jsoo version is 6.0 or higher. *)
+  val to_flags : recent:bool -> t -> string list
 end = struct
+  type effects_backend =
+    | Cps
+    | Double_translation
```

Comment:
Maybe we should add `Native` and `JSPI` so that we don't have to modify `dune` if we add support for native effects in wasm_of_ocaml.

Reply:
So dune would support an `--effects=native` flag for wasm_of_ocaml which may one day be supported by wsoo itself?

Reply:
Right

Reply:
The downside is that it increases the number of dune rules registered in dune. One rule per lib per config in `Config.all`. We currently have 27 config on master.  Having cps, double_translate, jspi, and native (what is native btw ?) and "none" would result in 45 configs. I really which we eventually come up with a better design with lazy loading of rules. That would unblock https://github.com/ocaml/dune/pull/7389.